### PR TITLE
docs: add Narwhals.dependencies to API reference

### PR DESCRIPTION
**Pull request by MarcoGorelli**
I think these can be useful to library devs

As much as we like dataframe-agnostic code, sometimes specialised library paths are necessary, and utils like these can help with that

## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue # 
- Closes #

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

### CODE REVIEW

File: docs/api-reference/dependencies.md
Comment (DeaMariaLeon) [inline docs/api-reference/dependencies.md:19]: Is `is_pandas_dataframe` twice??

Comment (MarcoGorelli) [reply docs/api-reference/dependencies.md]: oops thanks!

**Review by FBruzzesi — APPROVED** [review]
@MarcoGorelli  I know you were waiting for an approval GIF 😂

KittyCatGIF

**Pull request merged**
